src/addons/sourcemod/scripting/zr/config.inc

```diff
@@ -222,21 +231,66 @@ void ConfigLoad()
 		{
 			mapname[i] = CharToLower(mapname[i]);
 		}
-		Format(mapconfig, sizeof(mapconfig), "sourcemod/zombiereloaded/%s.cfg", mapname);
-		Format(path, sizeof(path), "cfg/%s", mapconfig);
+
+		if (bSMPath)
+		{
+			Format(mapconfig, sizeof(mapconfig), "configs/zr/maps/%s.cfg", mapname);
+			BuildPath(Path_SM, path, PLATFORM_MAX_PATH, mapconfig);
+			Format(path, sizeof(path), "addons/sourcemod/%s", mapconfig);
+		}
+		else
+		{
+			Format(mapconfig, sizeof(mapconfig), "sourcemod/zombiereloaded/%s.cfg", mapname);
+			Format(path, sizeof(path), "cfg/%s", mapconfig);
+		}
 
 		// Check if the new file path exist.
 		cfgexists = FileExists(path);
 		// File doesn't exist, then stop.
 		if (!cfgexists)
 			return;
 	}
-	
-	// Execute config file.
-	ServerCommand("exec %s", mapconfig);
-	
-	// Log action.
-	LogEvent(false, LogType_Normal, LOG_CORE_EVENTS, LogModule_Config, "Map Configs", "Executed map config file: %s", path);
+
+	// Execute config file - Log action if successful.
+	if (LoadConfigAndExec(path, bSMPath))
+		LogEvent(false, LogType_Normal, LOG_CORE_EVENTS, LogModule_Config, "Map Configs", "Executed map config file: %s", path);
+}
+
+/**
+ * Workaround to execute file from game/cfgs
+ * Or from addons/sourcemo/configs/zr/maps
+ */
+bool LoadConfigAndExec(char[] filepath, bool bSMPath = false)
+{
+	if (!bSMPath)
+	{
+		ServerCommand("exec %s", filepath);
+		return true;
+	}
+
+	char line[PLATFORM_MAX_PATH];
+	File hFile = OpenFile(filepath, "r");
+
+	// Check if the file was opened successfully
+	if (hFile == null)
+		return false;
+
+	// Read each line and execute it
+	while (ReadFileLine(hFile, line, sizeof(line)))
+	{
+		// Remove any newline or whitespace characters
+		TrimString(line);
+
+		// Skip empty lines or comments
+		if (line[0] == '\0' || StrContains(line, "//") == 0)
```

Comment:
ok for me but be careful if we have lines like this it will ignore the line:
```bash
exec test.cfg // Executes test config file
```

Reply:
It should be good now, even if un-tested.